src/Nethermind/Nethermind.JsonRpc/Modules/Trace/TraceRpcModule.cs

```diff
@@ -148,12 +148,18 @@ public ResultWrapper<ParityTxTraceFromReplay> trace_replayTransaction(Hash256 tx
             }
 
             Block block = blockSearch.Object!;
-            if (!stateReader.HasStateForBlock(block.Header))
+            SearchResult<BlockHeader> parentSearch = blockFinder.SearchForHeader(new BlockParameter(block.Header.ParentHash));
```

Comment:
is this because of the ForceSameBlock change?

Reply:
That one is at https://github.com/NethermindEth/nethermind/pull/9021/files#diff-8ededae48b6899e82e6e8d8d0bfba084ef969d0545d411eb2bb052a2e89c76edR175. 

Previously `BranchProcessor` set the state to the parent, its just not obvious from here.

Reply:
yeah, was referring to that